### PR TITLE
Fix serializedobject append for excessively large bytes length

### PR DESCRIPTION
**Pull request by wltsmrz**
`Ledger.from_json(ledger).calc_tx_hash()` breaks with `RangeError: Maximum call stack size exceeded` due to transaction that has 1311 AffectedNodes

Ledger index: 7501326
Transaction hash: 104514626FFB561440700F1130A9B0004DAD872AD6FBBCCD96D06AF6D4D50B11
See also: https://code.google.com/p/v8/issues/detail?id=2896

### CODE REVIEW

File: src/serializedobject.js [hunk 1]
```diff
@@ -140,10 +140,9 @@ SerializedObject.prototype.append = function(bytes) {
   // Make sure both buffer and bytes are Array. Either could potentially be a
   // Buffer.
   if (Array.isArray(this.buffer) && Array.isArray(bytes)) {
-    // Array::concat is horribly slow where buffer length is 100 kbytes + One
-    // transaction with 1100 affected nodes took around 23 seconds to convert
-    // from json to bytes.
-    Array.prototype.push.apply(this.buffer, bytes);
+    for (var i = 0; i < bytes.length; i++) {
```
Comment (sublimator) [inline src/serializedobject.js]: So there is a limit to the length of an arguments array you can apply

Comment (clark800) [inline src/serializedobject.js]: We should put a comment here explaining why this helps. If I saw this code outside the context of this pull request, I would probably refactor it back to the way it was before.

**Comment by clark800** [comment]
LGTM

**Comment by geertweening** [comment]
needs rebase

**Comment by geertweening** [comment]
@wltsmrz rebase please

**Comment by sublimator** [comment]
LGTM

**Pull request merged**